sly.el

```diff
@@ -5250,6 +5250,32 @@ Full list of frame-specific commands:
 
 
 ;;;;; SLY-DB buffer creation & update
+
+(defcustom sly-db-focus-debugger 'always
+  "Control how debugger windows are displayed.
+ALWAYS: the debugger window is always focused.
+NEVER: the debugger window is never focused.
+REPL: only at the REPL."
+  :group 'sly-debugger
+  :type '(choice (const always)
+                 (const never)
+                 (const repl)))
+
+(defun sly-db-should-focus-debugger-p ()
+  "Decide whether to select the debugger window.
+Behavior depends on the current value of
+`sly-db-focus-debugger'."
+  (cl-ecase sly-db-focus-debugger
+    (always t)
+    (never nil)
+    (repl
+     (when-let (win (selected-window))
+       (when-let (buf (window-buffer win))
+         (and (eq (buffer-local-value 'major-mode buf)
```

Comment:
It's easier and more idiomatic to use `with-current-buffer` instead of asking for `buffer-local-value` on every var.